app/scripts/services/authorization.js

```diff
@@ -0,0 +1,94 @@
+'use strict';
+
+angular.module("openshiftConsole")
+  .factory("AuthorizationService", function($q, Logger, $interval, DataService, APIService){
+    function AuthorizationService() {
+      this.rules = null;
+      this.intervalTimer = null;
+      this.forceReload = false;
+      this.canICreateSomeResource = false;
+      this.projectNameRules = null;
+    }
+
+    var normalizeRules = function(rules) {
+      var normalizedRules = {};
+      _.each(rules, function(rule) {
+        _.each(rule.resources, function(resource) {
+          normalizedRules[resource] = rule.verbs;
+        });
+      });
+      return normalizedRules;
+    };
+
+    AuthorizationService.prototype.getProjectRules = function(projectName) {
+      var self = this;
+      if (!self.rules || projectName !== self.projectNameRules) {
+        // Making an API call to get the rules if:
+        // - no rules are cached
+        // - when need to force update the rules
+        // - when switching between projects
+        Logger.log("AuthorizationService, loading user rules");
+        console.log("AuthorizationService, loading user rules");
+        if (self.forceReload) {
+          self.setForceReload(false);
+        }
+        if (!self.intervalTimer) {
+          // Force rules reload every 3 minutes
+          self.intervalTimer = $interval(function () {
+            self.setForceReload(true);
+          }, 180000);
+        }
+        self.projectNameRules = projectName;
+        var object = {kind: "SelfSubjectRulesReview",
+                      apiVersion: "v1"
+                    };
+        DataService.create('selfsubjectrulesreviews', null, object, {namespace: projectName}).then(
+          function(data) {
+            self.rules = normalizeRules(data.status.rules);
+          }, function(data) {
+            self.rules = null;
+        });
+      } else {
+        // Using cached data.
+        Logger.log("AuthorizationService, using cached rules");
+        self.rules = self.getRules();
+      }
+    };
+
+    // Method to determine if user can perform specified action on a resource.
+    // If resource is not specified, all resources will be checked if the action
+    // can be performed.
+    AuthorizationService.prototype.canI = function(resource, verb) {
```

Comment:
Do you have an example of how we'd use the "canI" function without specifying a resource? Checking you have that verb on all resources you have rules for (which likely is not all resources that exist) is odd.